[14.0][FIX] Unittests base_delivery_carrier_label: Separate github test run

mt-software-de opened the pull request:
Since base_delivery_carrier_label introduced a logic,
to a default stock.quant.package on stock.move.lines.
We always have to deal with it, in other not dependent addons. 
The context key `set_default_package` is used in multiple modules which do not have a dependency to base_delivery_carrier_label

mt-software-de commented:
cc @chienandalu @jbaudoux
This one is needed for https://github.com/OCA/delivery-carrier/pull/710

jbaudoux commented:
@mt-software-de Can you drop generated files from your commits? It makes migration easier, preventing cherry-pick conflicts

rousseldenis commented:
@mt-software-de Have you looked at https://github.com/OCA/delivery-carrier/pull/643 ?

I've externalized the automatic package creation in order to not depend on that on base module (in specific ones if needed)

mt-software-de committed:
[FIX] Unittests base_delivery_carrier_label: Separate github test run

mt-software-de committed:
[FIX] delivery_cttexpress: Delivery without package

mt-software-de committed:
[IMP] delivery_package_fee: Remove context set_default_package

mt-software-de committed:
[IMP] delivery_package_number: Remove context set_default_package

mt-software-de committed:
[FIX] delivery_schenker: Do not use number_of_packages

Otherwise in a scenario with a transfer including moves with packages
and without the wrong amount of pieces is send
Therefore also the dependency to delivery_package_number can be removed

mt-software-de commented:
> @mt-software-de Can you drop generated files from your commits? It makes migration easier, preventing cherry-pick conflicts

done

mt-software-de commented:
Can we merge this one?

mt-software-de commented:
> Can we merge this one?

ping @jbaudoux, @rousseldenis is it possible to merge this one?
Several PR's waiting on this. 

https://github.com/OCA/delivery-carrier/pull/710
https://github.com/OCA/delivery-carrier/pull/782
~https://github.com/OCA/delivery-carrier/pull/801~

mt-software-de commented:
Ping @jbaudoux @rousseldenis 
@hildickethan-S73

dreispt commented:
This PR i blocking a few others.
Can anyone care to review please?

mt-software-de commented:
@OCA/logistics-maintainers ?

hildickethan commented on delivery_schenker/models/delivery_carrier.py:
```diff
@@ -429,15 +432,15 @@ def _schenker_shipping_information_without_packages(self, picking):
                 "cargoDesc": picking.name,
                 # For a more complex solution use packaging properly
-                "grossWeight": self._schenker_shipping_information_round_weight(
-                    weight / picking.number_of_packages
-                ),
+                "grossWeight": self._schenker_shipping_information_round_weight(weight),
                 "volume": self._schenker_shipping_information_round_volume(
-                    self._schenker_shipping_information_without_packages_volume(picking)
+                    self._schenker_shipping_information_without_packages_volume(
+                        move_lines_without_package
+                    )
                 ),
                 "packageType": self.schenker_default_packaging_id.shipper_package_code,
                 "stackable": self.schenker_default_packaging_id.schenker_stackable,
-                "pieces": picking.number_of_packages,
+                "pieces": 1,
```
do you really need to remove the dependency for this module's tests to work?

mt-software-de replied:
Yes because without the module `base_delivery_carrier_label` installed before there is no package, so `number_of_packages` is 0. 
Also this part of code is not correct, because in a scenario where some moves are packed and one isn't, this will send the wrong amount for pieces. Here in this method always `1` needs to be the value for pieces.

mt-software-de commented:
One Last try Ping @jbaudoux? :-)

simahawk commented on .github/workflows/test.yml:
```diff
@@ -37,9 +37,18 @@ jobs:
           - container: ghcr.io/oca/oca-ci/py3.6-ocb14.0:latest
             name: test with OCB
             makepot: "true"
+            exclude: "base_delivery_carrier_label,delivery_carrier_label_batch,delivery_roulier,delivery_roulier_option,delivery_roulier_laposte_fr,delivery_roulier_chronopost_fr,carrier_account_environment"
```
Wait, this is going to slow down builds a lot.
Don't we have any better way to handle this? Normally this happens because modules tend to enable features by default. Can't we don anything to prevent that?

mt-software-de replied:
Yes but there is No Progress and No Response. See https://github.com/OCA/delivery-carrier/pull/643

mt-software-de replied:
Could we please merge this for now. I really waiting on this. If there is anyone which refactors the false implemented addon base_delivery_carrier_label, also the test flow can be changed again.

mt-software-de replied:
ping
Also https://github.com/OCA/delivery-carrier/pull/643 will created separate github flow runs.

hparfr replied:
@mt-software-de  Please have a look to https://github.com/OCA/delivery-carrier/pull/854
